src/main/java/edu/kis/powp/jobs2d/command/visitor/VisitorFlipX.java

```diff
@@ -1,31 +1,13 @@
 package edu.kis.powp.jobs2d.command.visitor;
 
-import edu.kis.powp.jobs2d.command.DriverCommand;
-import edu.kis.powp.jobs2d.command.ICompoundCommand;
-import edu.kis.powp.jobs2d.command.OperateToCommand;
-import edu.kis.powp.jobs2d.command.SetPositionCommand;
-
-import java.util.Iterator;
-
-public class VisitorFlipX implements VisitorCommand {
-
+public class VisitorFlipX extends TransformationVisitor {
 	@Override
-	public void visit(OperateToCommand operateToCommand) {
-		operateToCommand.setPosY(operateToCommand.getPosY() * (-1));
+	int calculateNewPositionX(int oldX, int oldY) {
+		return oldX;
 	}
```

Comment:
Such transformations doesn't need to be a part of a visitor subclass. You can use a strategy (or bridge) approach to gain new reusable components.